code/game/atoms_movable.dm

```diff
@@ -518,3 +517,39 @@
 	var/turf/currentturf = get_turf(src)
 	if(currentturf && (currentturf.z == ZLEVEL_CENTCOM || currentturf.z == ZLEVEL_STATION))
 		. = TRUE
+
+
+/* Language procs */
+/atom/movable/proc/grant_language(datum/language/dt)
+	if(!has_language(dt))
+		var/datum/language/L = new dt
+		LAZYADD(languages, L)
+
+/atom/movable/proc/grant_all_languages(ignore_restrictions=FALSE)
+	for(var/la in subtypesof(/datum/language))
+		grant_language(la)
+
+	if(ignore_restrictions)
+		SET_SECONDARY_FLAG(src, CAN_ALWAYS_SPEAK_A_LANGUAGE)
+
+/atom/movable/proc/remove_language(datum/language/dt)
+	if(languages && languages.len)
+		for(var/L in languages)
+			if(istype(L, dt))
+				languages -= L
+				qdel(L)
+				break
+
+/atom/movable/proc/has_language(datum/language/dt)
+	. = FALSE
+	if(languages)
+		for(var/L in languages)

```

Comment:
well... it's a list, they can't have new procs.
But you're welcome to make a new global proc (like ```typecahceof()``` or ```typecache_filter_list()```)

```typecache_filter_list()``` only exists because MSO needed to make it, and so he made it, feel free to add these kinds of "extension procs" for typecaches where you see value in them.